components/calendar/src/provider.rs

```diff
@@ -153,14 +153,200 @@ pub struct WeekDataV1 {
 }
 
 /// Bitset representing weekdays that are part of the 'weekend'.
+///
+/// Returns days via [`Self::days()`] method which returns [WeekendDays], an [Iterator].
+/// Users of this data do not need to interact with this Bitset,
+/// you can access [WeekendDays] using [`WeekDataV2::weekend()`].
+///
+/// # Internal representation
+///
+/// This Bitset uses an [u8] to represent the weekend, thus leaving one bit free.
+/// Each bit represents a day in the following order:
+///
+///   ┌▷Mon
+///   │┌▷Tue
+///   ││┌▷Wed
+///   │││┌▷Thu
+///   ││││ ┌▷Fri
+///   ││││ │┌▷Sat
+///   ││││ ││┌▷Sun
+///   ││││ │││
+/// 0b0000_1010
+///
+/// Please note that this is not a range, this are the discrete days representing a weekend. Other examples:
+/// 0b0101_1000 -> Tue, Thu, Fri
+/// 0b0000_0110 -> Sat, Sun
 #[derive(Clone, Copy, Debug, PartialEq)]
 #[cfg_attr(
     feature = "datagen",
-    derive(serde::Serialize, databake::Bake),
+    derive(databake::Bake),
     databake(path = icu_calendar::provider),
 )]
-#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
-pub struct WeekendSet(pub u8);
+pub struct WeekendSet(u8);
+
+impl WeekendSet {
+    /// Creates a new [WeekendSet] using the two provided days.
+    ///
+    /// Because of the current [CLDR spec](https://www.unicode.org/reports/tr35/tr35-dates.html#Date_Patterns_Week_Elements),
+    /// these are not a range, these are the two days representing a weekend.
+    /// If days are equal, weekend only has one day.
+    pub fn new(start_day: IsoWeekday, end_day: IsoWeekday) -> Self {
```

Comment:
nit: these should then be called `day1` and `day2`

Reply:
Renamed to `first_day` and `second_day` in https://github.com/unicode-org/icu4x/pull/4364/commits/c14d1b67f6f2db900330e543e585e6da79d19357
Let me know if you think we should still go with `day1` and `day2`